packages/codegen-ui-react/lib/forms/form-renderer-helper/display-value.ts

```diff
@@ -13,96 +13,171 @@
   See the License for the specific language governing permissions and
   limitations under the License.
  */
-import { FieldConfigMetadata, isValidVariableName } from '@aws-amplify/codegen-ui';
+import {
+  FieldConfigMetadata,
+  InvalidInputError,
+  isValidVariableName,
+  StudioFormValueMappings,
+} from '@aws-amplify/codegen-ui';
 import { StudioFormInputFieldProperty } from '@aws-amplify/codegen-ui/lib/types/form/input-config';
-import { Expression, factory, JsxAttribute, PropertyAssignment, SyntaxKind, NodeFlags } from 'typescript';
+import {
+  Expression,
+  factory,
+  JsxAttribute,
+  PropertyAssignment,
+  SyntaxKind,
+  NodeFlags,
+  CallExpression,
+} from 'typescript';
 import {
   buildBindingExpression,
   buildConcatExpression,
   isBoundProperty,
   isConcatenatedProperty,
 } from '../../react-component-render-helper';
 import { getRecordsName } from './form-state';
+import { getElementAccessExpression } from './invalid-variable-helpers';
 import { isModelDataType } from './render-checkers';
 
 export const getDisplayValueObjectName = 'getDisplayValue';
 
 /**
-    example:
-    suggestions={Array.from(authorRecords).map(([key, record]) => ({
-        id: key,
-        label: getDisplayValue['primaryAuthor']?.(record) ?? key,
-    }))}
+  authorRecords.map((r) => ({
+    id: r.id,
+    label: r.id,
+  }))
+ */
+
+/**
+  examples:
+  for model -
+  authorRecords.map((r) => ({
+    id: r.id,
+    label: getDisplayValue['primaryAuthor']?.(r) ?? r.id,
+  }))
+
+  for scalar - 
+  authorRecords.map((r) => ({
+    id: r.id,
+    label: r.id,
+  }))
  */
-export function getAutocompleteSuggestionsPropFromModel({
+function getModelTypeSuggestions({
   modelName,
   fieldName,
+  key,
+  isModelType,
 }: {
   modelName: string;
   fieldName: string;
-}): JsxAttribute {
-  return factory.createJsxAttribute(
-    factory.createIdentifier('suggestions'),
-    factory.createJsxExpression(
-      undefined,
-      factory.createCallExpression(
-        factory.createPropertyAccessExpression(
-          factory.createCallExpression(
-            factory.createPropertyAccessExpression(factory.createIdentifier('Array'), factory.createIdentifier('from')),
-            undefined,
-            [factory.createIdentifier(getRecordsName(modelName))],
-          ),
-          factory.createIdentifier('map'),
+  key: string;
+  isModelType: boolean;
+}): CallExpression {
+  const recordString = 'r';
+
+  const labelExpression = isModelType
+    ? factory.createBinaryExpression(
+        factory.createCallChain(
+          getElementAccessExpression('getDisplayValue', fieldName),
+          factory.createToken(SyntaxKind.QuestionDotToken),
+          undefined,
+          [factory.createIdentifier('r')],
```

Comment:
Nit: This 'r' should be `recordString` also?

Reply:
right, will bundle in next PR